tests/canary/scripts/push_stats_to_cloudwatch.py

```diff
@@ -0,0 +1,109 @@
+xml_path = "../canary/integration_tests.xml"
+
+def readXML_and_publish_metrics_to_cw():
+    if os.path.isfile(xml_path):
+        tree = ET.parse(xml_path)
+        testsuite = tree.find('testsuite')
+        failures = testsuite.attrib['failures']
+        errors = testsuite.attrib['errors']
+        tests = testsuite.attrib['tests']
+        success_rate = 1 - (int(failures)/int(tests))
+    else:
+        failures = 0
+        errors = 0
+        tests = 0
+        success_rate = 0
+    
+    timestamp = datetime.now().strftime("%Y-%m-%dT%H:%M:%S")
+
+
+    print (f"Failures: {failures}")
+    print (f"Errors: {errors}")
+    print (f"Total tests: {tests}")
+    print (f"Success_rate: {success_rate}")
+
+    #push to cloudwatch
+    cw_client = boto3.client("cloudwatch")
+    codebuild_client = boto3.client("codebuild")
+
+    codebuild_projects = codebuild_client.list_projects()['projects']
+    print(codebuild_projects)
+    project_name = codebuild_projects[0]
+
+    # Define the metric data
+    metric_data = [
+        {
+            'MetricName': 'failures',
+            'Timestamp': timestamp,
+            'Dimensions': [
+                {'Name': "CodeBuild Project Name", "Value": project_name},
+            ],
+            'Value': int(failures),
+            'Unit': 'Count'
+        },
+        {
+            'MetricName': 'errors',
+            'Timestamp': timestamp,
+            'Dimensions': [
+                {'Name': "CodeBuild Project Name", "Value": project_name},
+            ],
+            'Value': int(errors),
+            'Unit': 'Count'
+        },
+        {
+            'MetricName': 'total_tests',
+            'Timestamp': timestamp,
+            'Dimensions': [
+                {'Name': "CodeBuild Project Name", "Value": project_name},
+            ],
+            'Value': int(tests),
+            'Unit': 'Count'
+        },
+        {
+            'MetricName': 'successes_rate',
+            'Timestamp': timestamp,
+            'Dimensions': [
+                {'Name': "CodeBuild Project Name", "Value": project_name},
+            ],
+            'Value': success_rate,
+            'Unit': 'Percent'
+        }
+    ]
+
+# Use the put_metric_data method to push the metric data to CloudWatch
+    try:
+        response = cw_client.put_metric_data(
+            Namespace='Canary_Telemetry',
+            MetricData=metric_data
+        )
+        if response['ResponseMetadata']['HTTPStatusCode'] == 200:
+            print("Successfully pushed data to CloudWatch")
+            # return 200 status code if successful
+            return 200
+        else:
+            # raise exception if the status code is not 200
+            raise Exception("Unexpected response status code: {}".format(response['ResponseMetadata']['HTTPStatusCode']))
+    except Exception as e:
+        print("Error pushing data to CloudWatch: {}".format(e))
+        # raise exception if there was an error pushing data to CloudWatch
+        raise
+
```

Comment:
format using black. couple of extra lines

Reply:
formatted with black